### PR TITLE
[IMP] l10n_nl: Added tags needed for PL and BS tags reports

### PR DESCRIPTION
**Pull request by hsal-odoo**
In the Netherlands localization, the Balance Sheet and Profit&Loss reports are only currently built on account codes. We want to add another version of the accounting reports that uses account tags.

Enterprise PR: odoo/enterprise#58487
Task link: https://www.odoo.com/web#model=project.task&id=3794536
task-3794536

### CODE REVIEW

**Comment by rugo-odoo** [comment]
Can you provide a task-id for the pr ? Also don't forget to write a pr message.

File: addons/l10n_nl/data/account_account_tag.xml [hunk 1]
```diff
@@ -136,5 +136,67 @@
             <field name="applicability">accounts</field>
         </record>
 
+        <record id="account_tag_immat" model="account.account.tag">
+            <field name="name">Intangible Fixed Asset</field>
+            <field name="applicability">accounts</field>
```
Comment (malb-odoo) [inline addons/l10n_nl/data/account_account_tag.xml:141]: I think we should also add the country id here, to restrict the tag to the localisation

**Review by malb-odoo — COMMENTED** [review]
Small little thing i think we should add here :smile:

**Review by malb-odoo — COMMENTED** [review]
Thaaanks for the pr :smile:

File: addons/l10n_nl/i18n/de.po [hunk 2]
```diff
@@ -89,12 +103,19 @@ msgstr "3a. Lieferungen in Länder außerhalb der EU (Exporte) (Umsatz)"
...
+"3b. Lieferungen an/Dienstleistungen in Ländern innerhalb der EU und A-B-C-"
+"Lieferkettengeschäfte (Umsatz)"
 
 #. module: l10n_nl
-#: model:account.report.line,name:l10n_nl.tax_report_rub_3bl
-msgid "3bl. Deliveries to/services in countries within the EU (turnover)"
-msgstr "3bl. Lieferungen an/Dienstleistungen in Länder innerhalb der EU (Umsatz)"
+#: model:account.report.line,name:l10n_nl.tax_report_rub_3bg
+msgid "3bg. Leveringen producten naar landen binnen de EU (omzet)"
+msgstr ""
```
[the remark this answers is not in the document]
Comment (hsal-odoo) [reply addons/l10n_nl/i18n/de.po:113]: Should that be in a separate task? Or how do u suggest I fix it?

[the remark this answers is not in the document]
Comment (malb-odoo) [reply addons/l10n_nl/i18n/de.po:113]: Maybe just another commit would be ok

File: addons/l10n_nl/data/account_account_tag.xml [hunk 3]
```diff
@@ -3,138 +3,249 @@
     <data noupdate="0">
 
         <!-- Account Tags -->
-        <record id="account_tag_1" model="account.account.tag">
+        <record id="account_tag_1" model="account.account.tag"> <!-- Needed for PL tags report -->
```
Comment (malb-odoo) [inline addons/l10n_nl/data/account_account_tag.xml:6]: i don't think the comments are useful, it will just rewrite the github history

**Review by malb-odoo — COMMENTED** [review]
Small thing

File: addons/l10n_nl/i18n/de.po (hunk 2 above)
[the remark this answers is not in the document]
Comment (hsal-odoo) [reply addons/l10n_nl/i18n/de.po]: @malb-odoo done :+1:

**Review by malb-odoo — APPROVED** [review]
LGTM Thankssss
cc: @william-andre

File: addons/l10n_nl/models/account_chart_template.py [hunk 4]
```diff
@@ -9,5 +9,9 @@ class AccountChartTemplate(models.AbstractModel):
     def _post_load_data(self, template_code, company, template_data):
         super()._post_load_data(template_code, company, template_data)
         if template_code == 'nl':
-            company.account_journal_suspense_account_id.tag_ids += self.env.ref('l10n_nl.account_tag_25')
-            company.get_unaffected_earnings_account().tag_ids += self.env.ref('l10n_nl.account_tag_12')
+            cash_tag = self.env.ref('l10n_nl.account_tag_25', raise_if_not_found=False)
```
Comment (william-andre) [inline addons/l10n_nl/models/account_chart_template.py:12]: Does it really work to do the `+=` if it is not found?

Comment (hsal-odoo) [reply addons/l10n_nl/models/account_chart_template.py:12]: In what case would it be not found? Would not the tag be always there at CoA loading time?

File: addons/l10n_nl/migrations/3.4/post-migrate.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    for company in env['res.company'].search([('chart_template', '=', 'nl')]):
```
Comment (william-andre) [inline addons/l10n_nl/migrations/3.4/post-migrate.py:6]: To follow a fix made recently on all the similar scripts
```suggestion
    for company in env['res.company'].search([('chart_template', '=', 'nl')], order="parent_path"):
```

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Comment by william-andre** [comment]
@robodoo rebase-ff

**Pull request closed without merging**